src/analysis/processing/qgsalgorithmflattenrelationships.cpp

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+                         qgsalgorithmflattenrelationships.h
+                         ---------------------
+    begin                : August 2020
+    copyright            : (C) 2020 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmflattenrelationships.h"
+#include "qgsrelationmanager.h"
+#include "qgsvectorlayer.h"
+#include "qgsvectorlayerfeatureiterator.h"
+
+///@cond PRIVATE
+
+QString QgsFlattenRelationshipsAlgorithm::name() const
+{
+  return QStringLiteral( "flattenrelationships" );
```

Comment:
@nyalldawson based on the changes done in the PR due to review, should this instead be _flattenrelationship_? Or too late or you expect it addresses all relations one day?
Also the alg allows to skip the output, [optional]. Intended?